basetest.pm

```diff
@@ -291,13 +280,29 @@ sub runtest {
         $self->post_run_hook();
     };
     if ($@ || ($self->{result} || '') eq 'fail') {
-        my $msg = "test $name " . ($@ ? 'died: ' . $@ : 'failed');
-        warn $msg;
-        # add a fail screenshot in case there is none
-        if ($@ && (!@{$self->{details}} || ($self->{details}->[-1]->{result} || '') ne 'fail')) {
-            my $result = $self->record_testresult('fail');
-            $self->_result_add_screenshot($result);
+        my $msg;
+        if ($@) {
+            $self->{result} = 'fail';
+            $msg = "test $name died $@";
+            # add a fail screenshot in case there is none
+            if (!@{$self->{details}} || ($self->{details}->[-1]->{result} || '') ne 'fail') {
+                my $result = $self->record_testresult('unk');
+                $self->_result_add_screenshot($result);
+            }
+            my $details = {result => 'fail'};
+
+            my $text_fn = $self->next_resultname('txt');
+            open my $fd, ">", bmwqemu::result_dir() . "/$text_fn";
+            print $fd "# Test died:\n$@\n";
+            close $fd;
+            $details->{text}  = $text_fn;
+            $details->{title} = 'DIE';
```

Comment:
Looking at the screen shot, I'm not sure we want openQA to scream 'DIE' on unsuspecting user. Too bad `$text_fn` won't fit, but what about putting there first few chars that fit and `...` to signalize hint text?